include/nil/crypto3/zk/snark/arithmetization/plonk/detail/lookup_table_definition.hpp

```diff
@@ -0,0 +1,166 @@
+                            for( std::size_t i = 0; i < table->get_table().size(); i++ ){
+                                if(constant_columns[i].size() < start_row + table->get_table()[i].size()){
+                                    constant_columns[i].resize(start_row + table->get_table()[i].size());
+                                    if( usable_rows_after < start_row + table->get_table()[i].size() ){
+                                        usable_rows_after = start_row + table->get_table()[i].size();
+                                    }
+                                }
+                                for( std::size_t j = 0; j < table->get_table()[i].size(); j++ ){
+                                    constant_columns[i][start_row + j] = table->get_table()[i][j];
+                                }
+                            }
+
+                            for( const auto &[subtable_name, subtable]:table->subtables ){
+                                // Create selector
+                                plonk_column<FieldType> selector_column(usable_rows_after, FieldType::value_type::zero());
+                                for(std::size_t k = subtable.begin; k <= subtable.end; k++){
+                                    selector_column[start_row + k] = FieldType::value_type::one();
+                                }
+
+                                std::string full_table_name = table->table_name + "/" + subtable_name;
+                                bp_lookup_tables[lookup_table_ids.at(full_table_name) - 1] = plonk_lookup_table<FieldType>(subtable.column_indices.size(), cur_selector_id);
+                                std::vector<plonk_variable<typename FieldType::value_type>> option;
+                                for( const auto &column_index:subtable.column_indices ){
+                                    plonk_variable<typename FieldType::value_type> var(
+                                        constant_columns_ids[column_index], 0, false, plonk_variable<typename FieldType::value_type>::column_type::constant
+                                    );
+                                    option.push_back(var);
+                                }
+                                bp_lookup_tables[lookup_table_ids.at(full_table_name) - 1].append_option(option);
+
+                                assignment.fill_selector(cur_selector_id, selector_column);
+                                selector_columns.push_back(selector_column);
+                                // Create table declaration
+                                table_index++;
+                                cur_selector_id++;
+                            }
+                            start_row += table->get_table()[0].size();
+                        }
+                        for(std::size_t i = 0; i < bp_lookup_tables.size(); i++){
+                            bp.add_lookup_table(bp_lookup_tables[i]);

```

Comment:
If bp_lookup_tables is not used, maybe here also bp.add_lookup_table(std::move(bp_lookup_tables[i]));